### ISSUE TITLE
SC2143: false positive with pgrep

### ISSUE THREAD

**Issue by volans-** [koalaman/shellcheck#328]
With a script like:

``` sh
if [[ -n "$(pgrep -f "myapp")" ]]; then
    echo "test"
fi
```

I got:
``` sh
In test.sh line 3:
if [[ -n "$(pgrep -f "myapp")" ]]; then
      ^-- SC2143: Instead of [ -n $(foo | pgrep bar) ], use foo | pgrep -q bar .
```

But ```pgrep``` is a different tool and the rationale of [SC2143](https://github.com/koalaman/shellcheck/wiki/SC2143) doesn't apply.
Moreover it doens't have the -q option, so the suggested fix doesn't' work.
Is it recognized by shellcheck as one of the ```*grep``` variants?

Of course is possible to make the same construct with ```pgrep``` too, with something like:
``` sh
if pgrep -f "myapp" > /dev/null 2>&1; then
	echo "test"
fi
```
but given that the rationale of SC2143 doesn't apply, the only advantage will be to avoid the sub-shell.

**Comment by koalaman** [comment]
ShellCheck naively thought that *grep was a grep equivalent. This worked for agrep, egrep, fgrep and zgrep, but not pgrep. Fixed in 8e554ae
